odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -0,0 +1,35 @@
+import os
+
+import astroid
+from pylint import checkers, interfaces
+
+
+class OdooBaseChecker(checkers.BaseChecker):
+    __implements__ = interfaces.IAstroidChecker
+    name = 'odoo'
+
+    msgs = {
+        'E8502': (
+            'Bad usage of _ method.',
+            'gettext-variable',
+            'See https://www.odoo.com/documentation/13.0/reference/translations.html#variables'
+        )
+    }
+
+    def _check_variable_inside_gettext(self, node):
+        if not (
+            isinstance(node, astroid.Call) and node.args and
+            isinstance(node.func, astroid.Name) and node.func.name in ('_', '_lt')
+        ):
+            return False
+        first_arg = node.args[0]
+        return not(isinstance(first_arg, astroid.Const) and isinstance(first_arg.value, str))
+
+    @checkers.utils.check_messages('gettext-variable')
+    def visit_call(self, node):
+        if self._check_variable_inside_gettext(node):
+            self.add_message('gettext-variable', node=node)
```

Comment:
The utility function seems unnecessary at this point given how small it is, it could be folded into `visit_call` (with the check on `astroid.Call` stripped) e.g.
```suggestion
    @checkers.utils.check_messages('gettext-variable')
    def visit_call(self, node):
        if isinstance(node.func, astroid.Name) and node.func.name in ('_', '_lt'):
            first_arg = node.args[0]
            if not (isinstance(first_arg, astroid.Const) and isinstance(first_arg.value, str)):
                self.add_message('gettext-variable', node=node)
```

Reply:
What about calls of `_`, `_lt` without args?  They could be flagged as well.

Reply:
> What about calls of `_`, `_lt` without args?

But the first argument is required, so you should get a TypeError anyway

Reply:
> What about calls of `_`, `_lt` without args? They could be flagged as well.

it's going to crash the linter, that probably counts as "flagged" :D